massa-protocol-worker/src/handlers/block_handler/retrieval.rs

```diff
@@ -503,14 +502,18 @@ impl RetrievalThread {
                     let mut endorsement_cache_write = self.endorsement_cache.write();
                     let endorsement_ids = endorsement_cache_write
                         .endorsements_known_by_peer
-                        .get_or_insert_mut(from_peer_id.clone(), || {
-                            LruCache::new(
-                                NonZeroUsize::new(self.config.max_node_known_blocks_size)
+                        .get_or_insert(from_peer_id.clone(), || {
+                            LruMap::new(ByLength::new(
+                                self.config
+                                    .max_node_known_blocks_size
+                                    .try_into()
                                     .expect("max_node_known_blocks_size in config must be > 0"),
-                            )
-                        });
+                            ))
+                        })
+                        .ok_or(())
+                        .expect("endorsements_known_by_peer limit reached");
```

Comment:
We shouldn't have any except because we can't save data in the cache. The cache is a good to have feature but shouldn't kill the whole handler if the cache is full. The except that was there before was on a config value set at start not being 0. I think we should remove all the except to change them to result with warning in case of error but don't stop the function

Reply:
Yes I just didn't place a comment each time but it apply everywhere.